File: 4.group_anagram/main.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const groupedString = new Map();

    for(const str of strs){
        const newArr = str.split('');
        newArr.sort();
        const sortedArr = newArr.join('');

        if(!groupedString.has(sortedArr)){
            groupedString.set(sortedArr, []);
        }
        groupedString.get(sortedArr).push(str);
    }
    return Array.from(groupedString.values());
}

File: 4.group_anagram/main.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> groupedString = new HashMap<>();
        for(String str: strs){
            char[] newArr = str.toCharArray();
            Arrays.sort(newArr);
            String sortedStr = String.valueOf(newArr);
            if(!groupedString.containsKey(sortedStr)){
                groupedString.put(sortedStr, new ArrayList<>());
            }
            groupedString.get(sortedStr).add(str);
        }
        return new ArrayList<>(groupedString.values());
    }
}